hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/DropHoodieTableCommand.scala

```diff
@@ -69,13 +67,13 @@ extends HoodieLeafRunnableCommand {
     val catalog = sparkSession.sessionState.catalog
 
     // Drop table in the catalog
-    val enableHive = isEnableHive(sparkSession)
-    if (enableHive) {
-      dropHiveDataSourceTable(sparkSession, hoodieCatalogTable)
+    if (HoodieTableType.MERGE_ON_READ == hoodieCatalogTable.tableType && purge) {
+      val (rtTableOpt, roTableOpt) = getTableRTAndRO(catalog, hoodieCatalogTable)
+      rtTableOpt.foreach(table => catalog.dropTable(table.identifier, true, false))
+      roTableOpt.foreach(table => catalog.dropTable(table.identifier, true, false))
+      catalog.dropTable(table.identifier.copy(table = hoodieCatalogTable.tableName), ifExists, purge)
     } else {
-      if (catalog.tableExists(tableIdentifier)) {
-        catalog.dropTable(tableIdentifier, ifExists, purge)
-      }
+      catalog.dropTable(table.identifier, ifExists, purge)
```

Comment:
`purge` is false  and `HoodieTableType.MERGE_ON_READ == hoodieCatalogTable.tableType` Don't need to drop `ro` and `rt` table?

Reply:
In existing code, RT and RO tables are dropped only when purging a MOR table; This PR respects the current behavior.
From my point, such logic makes sense and acceptable for the user.

Reply:
> In existing code, RT and RO tables are dropped only when purging a MOR table; This PR respects the current behavior. From my point, I also think such logic makes sense and acceptable for the user.

Will this cause the metadata of the `ro` and `rt` tables to remain? Causes failure when re-creating the mor table.

Reply:
I think it should be fine. The `ro` and `rt` is left with NO-PURGE, user can also drop them separately case by case.
Just like the test of `Test("Test Drop RO & RT table by one by one.")`

Reply:
> I think it should be fine. The `ro` and `rt` is left with NO-PURGE, user can also drop them separately case by case. Just like the test of `Test("Test Drop RO & RT table by one by one.")`

well